### PR TITLE
feat: Add sub-microsecond precision using floats for timer metrics

### PR DESCRIPTION
**Pull request by mythmon**

### CODE REVIEW

File: merino-suggest/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+use cadence::{Histogram, Histogrammed, MetricBuilder, MetricResult};
+use std::time::Duration;
+
+/// Trait for recording timer values with additional precision.
+///
+/// Time data is stored as a histogram.  Statistical distribution is calculated
+/// by the server. Times will be stored a number of milliseconds, with
+/// fractional values used to represent nanoseconds.
+///
+/// Only `Duration types are valid.
```
Comment (pjenvey) [inline merino-suggest/src/metrics.rs:12]:
```suggestion
/// Only `Duration` types are valid.
```

**Review by pjenvey — CHANGES_REQUESTED** [review]

**Comment by mythmon** [comment]
The check failure is `cargo audit`. I've made #220 to handle that.

**Review by pjenvey — APPROVED** [review]

**Comment by Dexterp37** [comment]
Do we need to update the data docs ?

**Comment by mythmon** [comment]
The data docs currently say

> A histogram that records the amount of time, in microseconds, that the adM Remote Settings provider took to generate suggestions.

That doesn't specify what the precision of those microseconds are, so I don't think we need to update the docs.

**Comment by mythmon** [comment]
Rebased to pick up CI changes.

**Pull request merged**